src/ray/raylet/local_object_manager.cc

```diff
@@ -253,15 +253,26 @@ void LocalObjectManager::SpillObjectsInternal(
                 num_active_workers_ -= 1;
               }
               io_worker_pool_.PushSpillWorker(io_worker);
-              if (!status.ok()) {
-                for (const auto &object_id : objects_to_spill) {
+              size_t num_objects_spilled = r.spilled_objects_url_size();
+              // Object spilling is always done in the order of the request.
+              // For example, if an object succeeded, it'll guarentee that all objects
+              // before this will succeed.
+              RAY_CHECK(num_objects_spilled <= objects_to_spill.size());
+              if (!status.ok() || num_objects_spilled < objects_to_spill.size()) {
+                auto failed_num = objects_to_spill.size() - num_objects_spilled;
+                RAY_LOG(ERROR) << "Fail to spill " << failed_num << " objects.";
```

Comment:
It looks like we'll end up printing the error twice because there was already an ERROR log for !status.ok. Could you consolidate the error messages?